Landing page

camifernweh opened the pull request.

camifernweh committed:
feat: add header

camifernweh committed:
style: update theme

camifernweh committed:
feat: add animated pictures

camifernweh committed:
feat: enable server-side rendering for styles, add responsive carousel

camifernweh committed:
afeat: add landing-cards component

camifernweh committed:
style: fix header spacing

camifernweh committed:
feat: integrate layout with landing page

camifernweh committed:
Merge branch 'main' into landingPage

TobiBecker commented on pages/index.tsx:
```diff
@@ -12,15 +13,17 @@ const Home = ({t, i18n}): React.ReactElement => {
-      </Button>
+      </Button> -->
+      
+      <LandingPage />
```
The commenting out didn't work here. Page was breaking.

TobiBecker committed:
fix: syntax for comment out translation button

TobiBecker commented on pages/_app.tsx:
```diff
@@ -1,18 +1,37 @@
 
-const MyApp = ({ Component, pageProps }: AppProps): React.ReactElement => {
+function MyApp({ Component, pageProps }: AppProps): React.ReactNode {
+  React.useEffect(() => {
+    // Remove the server-side injected CSS.
+    const jssStyles = document.querySelector('#jss-server-side');
+    if (jssStyles) {
+      jssStyles.parentElement.removeChild(jssStyles);
+    }
+  }, []);
```
can we not use useEffect instead of React.useEffect?

TobiBecker commented on containers/landing/LandingPage.tsx:
```diff
@@ -0,0 +1,18 @@
+
+const LandingPage = (): React.ReactElement => {
+  return (
+    <React.Fragment>
```
can we import Fragment  instead of using React.Fragment?

beckertobias commented on components/header/Header.tsx:
```diff
@@ -0,0 +1,78 @@
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import AnimatedPics from '../animated-pics/AnimatedPics';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      marginTop: '1rem',
+      marginBottom: '1rem',
+    },
+    textContainer: {
+      marginLeft: '3rem',
+    },
+    mainText: {
+      margin: '3rem 0 2rem 2rem',
+      fontSize: '2rem',
+      fontWeight: 'bold',
+    },
+    secondaryText: {
+      color: 'rgb(255, 255, 255)',
+      fontWeight: 'bold',
+      fontSize: '1.25rem',
+      lineHeight: '1.5',
+    },
```
We are probably not reusing these styles anywhere (I think you also mentioned that). But for the other components it will make sense to put more in the "global theme" and import from there.
Like having default fontSizes (small, medium, large) and colors etc. there.

atsteinberg replied:
and why is the styling for the whole page in the header component? Or is mainText etc. only for the header? Ah. I see. The 'header' is basically the main component of the page?

TobiBecker committed:
fix: comment out handleClick on index page

beckertobias commented on components/navbar/Navbar.tsx:
```diff
@@ -0,0 +1,142 @@
+        className={classes.middleGridContainer}
+      >
+        <Grid item>
+          <Link href="/">
+            <a className={classes.a}>
+              <Typography variant="h6" className={classes.link}>
+                Home
+              </Typography>
+            </a>
+          </Link>
```
Regarding all Link Components: 
- Forgot to say that after implementing the translation thing, we have to use a Link from "i18n" instead of next.js (because the route path needs to take of the locale (de / en )
- Alex created a Button component that can get an href. So it also returns a Link Component if you want.
- I changed our Button Component so that it uses the Link from i18n

--> We need to either import the Link from i18n and use it. Or maybe better: use our own Button component and pass it an href

beckertobias commented on pages/_app.tsx:
```diff
@@ -21,5 +40,9 @@ const MyApp = ({ Component, pageProps }: AppProps): React.ReactElement => {
 MyApp.getInitialProps = async (appContext: AppContextType<Router>) => ({ ...await App.getInitialProps(appContext) })
 
-
 export default appWithTranslation(MyApp);
+
+MyApp.propTypes = {
+  Component: PropTypes.elementType.isRequired,
+  pageProps: PropTypes.object.isRequired,
+};
```
Typescript now complains that the type of MyApp doesn't fit. Maybe because of the additional propTypes. But we can also fix that later...

atsteinberg replied:
Getting rid of propTypes and reverting line 15 to what it was before (i.e. using React.ReactElement instead of React.ReactNode fixes this

camifernweh committed:
fix: import link from i18next, minor react syntax changes

atsteinberg commented on components/animated-pics/AnimatedPics.tsx:
```diff
@@ -0,0 +1,51 @@
+import Carousel from 'react-material-ui-carousel';
+import Card from '@material-ui/core/Card';
+import CardMedia from '@material-ui/core/CardMedia';
```
I would suggest we use named imports from top-level component (i.e. @material-ui/core) and rely on webpack's tree-shaking capabilities to keep production bundle size down. In places you save quite a few lines of import statements this way.

atsteinberg commented on components/animated-pics/AnimatedPics.tsx:
```diff
@@ -0,0 +1,51 @@
+import Carousel from 'react-material-ui-carousel';
+import Card from '@material-ui/core/Card';
+import CardMedia from '@material-ui/core/CardMedia';
+import { makeStyles, createStyles } from '@material-ui/core/styles';
```
Maybe we should just decide once and for all on a styling strategy. Are there any up- or downsides to doing it this way (i.e. within the ts file rather than in a separate module)? For me, this way of doing tends to result in a bit too much styling within the components for my tastes (separation of concerns and all that), but that's the only concern I have.

atsteinberg commented on components/footer/Footer.tsx:
```diff
@@ -0,0 +1,35 @@
+const Footer = (): React.ReactElement => {
+  const classes = useStyles();
+
+  const year: string = new Date().getFullYear().toString();
```
haha. An auto-renewing copyright. That's not a thing, or is it?

atsteinberg commented on components/footer/Footer.tsx:
```diff
@@ -0,0 +1,35 @@
+        align="center"
+        className={classes.footer}
+      >
+        Kara {year} &copy;
```
isn't this the wrong way around?

atsteinberg commented on components/landing-cards/LandingCards.tsx:
```diff
@@ -0,0 +1,108 @@
+
+  return (
+    <Grid container className={classes.cardsContainer}>
+      {content.map((el) => {
```
I like the approach! Perhaps it would be slightly cleaner to define a const (e.g. contentGrids) before the return statement and simply use that within the tsx return

atsteinberg replied:
or even make a separate LandingCard component which is used by the LandingCards(List) component

atsteinberg commented on components/header/Header.tsx:
```diff
@@ -0,0 +1,78 @@
+      alignItems="center"
+      className={classes.root}
+    >
+      <Grid item lg={5} className={classes.textContainer}>
```
in the end, this should go through the t-Function. Actually, it's not a big deal to do that straight away rather than hard-coding the text snippets. But we should also think about a reasonable structure for our translation files before things get out of hand.

atsteinberg commented on components/navbar/Navbar.tsx:
```diff
@@ -0,0 +1,142 @@
+        spacing={1}
+        className={classes.middleGridContainer}
+      >
+        <Grid item>
```
These Grid item Link complexes should probably be made into their own little component (they get reused - with slight variation - 6 times). I know you don't share my love for components. But when you retype stuff more than twice, it's probably safe to say that code readability and maintainability would profit from moving it into its own file.

atsteinberg commented on pages/_app.tsx:
```diff
@@ -1,25 +1,50 @@
+import App from 'next/app';
 import { AppProps } from 'next/app';
-import { appWithTranslation } from "../i18n";
+import PropTypes from 'prop-types';
```
don't need it

atsteinberg commented on pages/_app.tsx:
```diff
@@ -1,25 +1,50 @@
+    // Remove the server-side injected CSS.
+    const jssStyles = document.querySelector('#jss-server-side');
+    if (jssStyles) {
+      jssStyles.parentElement.removeChild(jssStyles);
```
This misses a question mark, otherwise it throws a warning

atsteinberg reviewed (commented):
Landing page looks very cool. A few things should be fixed and I would strongly suggest moving the Link/Grid/Typography stuff into its own component. Perhaps we can also agree on a strategy to somehow separate the styling stuff from the functional bits (or at least agree that there is no good way of doing it). One minor thing: is there a way of perhaps cross-fading the different images. Right now the transition feels a tiny bit abrupt. 
But great job over all! This is quite some improvement to how it was before, wouldn't you say>

atsteinberg committed:
fix: remove propTypes

atsteinberg committed:
fix: remove propTypes

atsteinberg committed:
Merge branch 'landingPage' of https://github.com/fullstacksmart/next-kara into landingPage

camifernweh committed:
feat: add styles to separate files, refactor imports and add NavbarItems component

camifernweh committed:
pull remote changes
Merge branch 'landingPage' of https://github.com/fullstacksmart/next-kara into landingPage

camifernweh committed:
Merge branch 'main' into landingPage
